### PR TITLE
Implement ODK Entities for project creation

### PR DESCRIPTION
**Pull request by spwoodcock**
## What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation
- [ ] 🧑‍💻 Refactor
- [ ] ✅ Test
- [ ] 🤖 Build or CI
- [ ] ❓ Other (please specify)

## Related Issue

Related to https://github.com/hotosm/fmtm/issues/1356
Linked to https://github.com/hotosm/fmtm/issues/1319

## Describe this PR

Major update to how FMTM works:
- Instead of one form per task, we only need one form per building.
- An entity list is made per project.
- The entities can be filtered by the task_id field for display in ODK Collect.
- We can also launch a form in ODK Collect with a specific entity prefilled.
- We should be able to use the form to update the entities fields, e.g. a `status` field for when complete.

Things to do:
- Needs testing for project create and form update.
- I haven't updated any of the submissions logic (perhaps in another PR?)
- But now we can update to use the API on the form for the entire project.
- If you could please review & test it would be a great help @Sujanadh 🙏 

## Screenshots

Please provide screenshots of the change.

## Alternative Approaches Considered

Did you attempt any other approaches that are not documented in code?

## Review Guide

Notes for the reviewer. How to test this change?

## Checklist before requesting a review

- 📖 Read the FMTM Contributing Guide: <https://github.com/hotosm/fmtm/blob/main/CONTRIBUTING.md>
- 📖 Read the HOT Code of Conduct: <https://docs.hotosm.org/code-of-conduct>
- 👷‍♀️ Create small PRs. In most cases, this will be possible.
- ✅ Provide tests for your changes.
- 📝 Use descriptive commit messages.
- 📗 Update any related documentation and include any relevant screenshots.

### CODE REVIEW

File: src/backend/app/db/db_models.py [hunk 1]
```diff
@@ -277,6 +277,21 @@ class DbXForm(Base):
...
+    """XForms linked per project.
+
+    TODO eventually we will support multiple forms per project.
+    TODO So the category field a stub until then.
+    TODO currently it's maintained under projects.xform_category.
+    """
+
+    __tablename__ = "xforms"
+    id = cast(int, Column(Integer, primary_key=True, autoincrement=True))
+    project_id = cast(int, Column(Integer))
```
Comment (Sujanadh) [inline src/backend/app/db/db_models.py:290]: Foreign key here probably

File: src/backend/app/db/postgis_utils.py [hunk 2]
```diff
@@ -549,3 +550,73 @@ def get_address_from_lat_lon(latitude, longitude):
...
+
+    Args:
+        geojson_geometry (dict): The geojson polygon geom.
+
+    Returns:
+        str: A string representing the geometry in JavaRosa format.
+    """
+    # Ensure the GeoJSON geometry is of type Polygon
+    # FIXME support other geom types
+    if geojson_geometry["type"] != "Polygon":
```
Comment (Sujanadh) [inline src/backend/app/db/postgis_utils.py:574]: Got value error for some data extract which were point type

**Comment by spwoodcock** [comment]
A consequence of using Entities is we need to call the ODK Central API for every **feature** instead of every **task**, so this obviously adds a lot of calls!

We could merge this for now and project creation will take a long time.
It's quite high priority to make a PR to ODK Central to bulk upload Entities.

**Comment by spwoodcock** [comment]
Entity creation works now. It's pretty fast. I removed the project log to improve user experience.

**Comment by spwoodcock** [comment]
Merging this to dev. We can fix any potential issues after.
Still to do:
- Update the project detail page to use OdkEntity.getEntityData (uses odata endpoint).
- Replace using phonenumber in submissions with actual data fields from the Entities (https://github.com/hotosm/fmtm/issues/1425)
- https://github.com/hotosm/fmtm/issues/1417
- Just a thought: perhaps we don't even need to maintain our own data extract after using Entities?

**Pull request merged**